src/submit.rs

```diff
@@ -476,4 +477,70 @@ impl<'a> Submitter<'a> {
         )
         .map(drop)
     }
+
+    /// Performs a synchronous cancellation request, similiar to [AsyncCancel](crate::opcode::AsyncCancel),
+    /// except that it completes synchronously.
+    ///
+    /// Cancelation can target a specific request, or all requests matching a specific criteria. The
+    /// [MatchOn](types::MatchOn) builder supports describing the match criteria for cancelation.
+    ///
+    /// An optional `timeout` can be provided to specify how long to wait for matched requests to be
+    /// canceled. If no timeout is provided, the default is to wait indefinitely.
+    ///
+    /// ### Errors
+    ///
+    /// An error is returned if no requests match the provided match criteria. If a timeout is supplied,
+    /// and the timeout expires before all requests have been canceled, then an error will be returned.
+    ///
+    /// ### Notes
+    ///
+    /// Only requests which have been submitted to the ring will be considered for cancelation. Requests
+    /// which have been written to the SQ, but not submitted, will not be canceled.
+    ///
+    /// Available since 6.0.
+    pub fn register_sync_cancel(
+        &self,
+        match_on: types::MatchOn,
+        timeout: Option<Timespec>,
+    ) -> io::Result<()> {
+        let timespec = match timeout {
+            Some(ref ts) => {
+                // Safety: Timespec is repr(transparent) over __kernel_timespec, so the cast is safe.
+                unsafe { *(ts as *const Timespec as *const sys::__kernel_timespec) }
+            }
+            None => {
+                // Default to an infinite timeout if a timeout is not provided.
+                sys::__kernel_timespec {
+                    tv_sec: -1,
+                    tv_nsec: -1,
+                }
+            }
+        };
+
+        let flags = match_on.flags();
+        let user_data = match_on.user_data.unwrap_or(0);
+        let fd = match_on
+            .fd
+            .map(|fd| match fd {
+                types::sealed::Target::Fd(fd) => fd,
+                types::sealed::Target::Fixed(fd) => fd as i32,
```

Comment:
Maybe the register version of this can't cancel a fixed file descriptor? I will submit a question to the liburing repo.